code/modules/hydroponics/hydro_tray.dm

```diff
@@ -257,23 +257,14 @@
 		var/missing_gas = 0
 		for(var/gas in seed.consume_gasses)
 			if(environment)
-				switch(gas)
-					if("oxygen")
-						if(environment.oxygen <= seed.consume_gasses[gas])
-							missing_gas++
-							continue
-					if("plasma")
-						if(environment.toxins >= seed.consume_gasses[gas])
-							missing_gas++
-							continue
-					if("nitrogen")
-						if(environment.nitrogen >= seed.consume_gasses[gas])
-							missing_gas++
-							continue
-					if("carbon_dioxide")
-						if(environment.carbon_dioxide >= seed.consume_gasses[gas])
-							missing_gas++
-							continue
+				if(gas == OXYGEN)
```

Comment:
the fact that they use oxygen should probably be a var on the plants (or a define in this file)